client/post-editor/editor-ground-control/index.jsx

```diff
@@ -59,13 +64,34 @@ export default React.createClass( {
 		};
 	},
 
+	componentDidMount: function() {
+		if ( ! config( 'email_verification_gate' ) ) return;
+		
+		user.on( 'change', this.updateNeedsVerification );
+		user.on( 'verify', this.updateNeedsVerification );
+	},
+
+	componentWillUnmount: function() {
+		if ( ! config( 'email_verification_gate' ) ) return;
+		
+		user.off( 'change', this.updateNeedsVerification );
+		user.off( 'verify', this.updateNeedsVerification );
+	},
+
+	updateNeedsVerification: function() {
+		this.setState( {
+			needsVerification: config( 'email_verification_gate' ) && userUtils.needsVerificationForSite( this.props.site ),
+		} );
+	},
+
 	getInitialState: function() {
 		return {
 			showSchedulePopover: false,
 			showAdvanceStatus: false,
 			showDateTooltip: false,
 			firstDayOfTheMonth: this.getFirstDayOfTheMonth(),
-			lastDayOfTheMonth: this.getLastDayOfTheMonth()
+			lastDayOfTheMonth: this.getLastDayOfTheMonth(),
+			needsVerification: config( 'email_verification_gate' ) && userUtils.needsVerificationForSite( this.props.site ),
```

Comment:
As discussed in slack, using props in `getInitialState` can be an anti-pattern if the prop is updated after the component mounts. Until the user object is moved to redux, we discussed also updating the state on `componentWillReceiveProps`.